### ISSUE TITLE
Problem when I using Resolver Field?

### ISSUE THREAD

**Issue by siamahnaf198** [nestjs/nest#9752]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

Hello, I have a category resolver. In this resolver I write two query one for category and other for subcategory.

There entity is-

category.entity.ts-
```
@ObjectType()
export class Category {
    @Field(() => ID, { nullable: false })
    id: ObjectId;
    @Field(() => String, { nullable: false })
    name: string;
    @Field(() => String, { nullable: true })
    description: string;
    @Field(() => [Subcategory], { nullable: true })
    subCategory: Subcategory[];
    @Field(() => Date, { nullable: false })
    createdAt: DateScalar;
    @Field(() => Date, { nullable: false })
    updatedAt: DateScalar;
}
```

and subcategory.entity.ts

```
@ObjectType()
export class Subcategory {
    @Field(() => ID, { nullable: false })
    id: ObjectId;
    @Field(() => String, { nullable: false })
    name: string;
    @Field(() => Category, { nullable: false })
    category: Category;
    @Field(() => Date, { nullable: false })
    createdAt: DateScalar;
    @Field(() => Date, { nullable: false })
    updatedAt: DateScalar;
}
```

And this is resolver-

```
@Resolver(Category)
export class CategoryResolver {
    //Constructor
    constructor(private readonly categoryService: CategoryService) { }
    
    //Get All Category
    @Query(() => [Category], { name: "getCategories" })
    categories() {
        return this.categoryService.categories()
    }

    @ResolveField("subCategory", () => [Subcategory])
    subCategoryBatch(
        @Parent() category: Category
    ) {
        console.log(category)
    }



    //Get All Sub Category
    @Query(() => [Subcategory], { name: "getSubCategories" })
    subCategories() {
        return this.categoryService.subCategories();
    }
    @ResolveField("category", () => Category)
    categoryBatch(
        @Parent() subCategory: Subcategory
    ) {
        console.log(subCategory)
    }
}
```

Here we can see two resolver field. And I have to keep `Category` in @Resolver decorator. When I keep only `Category` , then it work for Category Resolve Field. Other is not working. Again when I keep `Subcategroy` in @Resolver decorator then it work for Subcategory Resolve Field, other is not working.

### Minimum reproduction code

https://github.com/siamahnaf198/ebuy-back

### Expected behavior

It should working.

### Package

- [ ] I don't know. Or some 3rd-party package
- [ ] <code>@nestjs/common</code>
- [ ] <code>@nestjs/core</code>
- [ ] <code>@nestjs/microservices</code>
- [ ] <code>@nestjs/platform-express</code>
- [ ] <code>@nestjs/platform-fastify</code>
- [ ] <code>@nestjs/platform-socket.io</code>
- [ ] <code>@nestjs/platform-ws</code>
- [ ] <code>@nestjs/testing</code>
- [ ] <code>@nestjs/websockets</code>
- [X] Other (see below)

### Packages versions

{
  "name": "server",
  "version": "0.0.1",
  "description": "",
  "author": "",
  "private": true,
  "license": "UNLICENSED",
  "scripts": {
    "prebuild": "rimraf dist",
    "build": "nest build",
    "format": "prettier --write \"src/**/*.ts\" \"test/**/*.ts\"",
    "start": "nest start",
    "start:dev": "nest start --watch",
    "start:debug": "nest start --debug --watch",
    "start:prod": "node dist/main",
    "lint": "eslint \"{src,apps,libs,test}/**/*.ts\" --fix",
    "test": "jest",
    "test:watch": "jest --watch",
    "test:cov": "jest --coverage",
    "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
    "test:e2e": "jest --config ./test/jest-e2e.json"
  },
  "dependencies": {
    "@nestjs-modules/mailer": "^1.6.1",
    "@nestjs/apollo": "^10.0.14",
    "@nestjs/common": "^8.0.0",
    "@nestjs/config": "^2.0.1",
    "@nestjs/core": "^8.0.0",
    "@nestjs/graphql": "^10.0.14",
    "@nestjs/mongoose": "^9.1.0",
    "@nestjs/platform-express": "^8.0.0",
    "@types/jsonwebtoken": "^8.5.8",
    "@types/nodemailer": "^6.4.4",
    "@types/otp-generator": "^4.0.0",
    "apollo-server-express": "^3.8.1",
    "axios": "^0.27.2",
    "bcrypt": "^5.0.1",
    "class-transformer": "^0.5.1",
    "class-validator": "^0.13.2",
    "dataloader": "^2.1.0",
    "google-auth-library": "^8.0.2",
    "graphql": "^16.5.0",
    "i": "^0.3.7",
    "jsonwebtoken": "^8.5.1",
    "mongoose": "^6.3.5",
    "nodemailer": "^6.7.5",
    "npm": "^8.12.1",
    "otp-generator": "^4.0.0",
    "reflect-metadata": "^0.1.13",
    "rimraf": "^3.0.2",
    "rxjs": "^7.2.0"
  },
  "devDependencies": {
    "@nestjs/cli": "^8.0.0",
    "@nestjs/schematics": "^8.0.0",
    "@nestjs/testing": "^8.0.0",
    "@types/express": "^4.17.13",
    "@types/jest": "27.5.0",
    "@types/node": "^16.0.0",
    "@types/supertest": "^2.0.11",
    "@typescript-eslint/eslint-plugin": "^5.0.0",
    "@typescript-eslint/parser": "^5.0.0",
    "eslint": "^8.0.1",
    "eslint-config-prettier": "^8.3.0",
    "eslint-plugin-prettier": "^4.0.0",
    "jest": "28.0.3",
    "prettier": "^2.3.2",
    "source-map-support": "^0.5.20",
    "supertest": "^6.1.3",
    "ts-jest": "28.0.1",
    "ts-loader": "^9.2.3",
    "ts-node": "^10.0.0",
    "tsconfig-paths": "4.0.0",
    "typescript": "^4.3.5"
  },
  "jest": {
    "moduleFileExtensions": [
      "js",
      "json",
      "ts"
    ],
    "rootDir": "src",
    "testRegex": ".*\\.spec\\.ts$",
    "transform": {
      "^.+\\.(t|j)s$": "ts-jest"
    },
    "collectCoverageFrom": [
      "**/*.(t|j)s"
    ],
    "coverageDirectory": "../coverage",
    "testEnvironment": "node"
  }
}

### In which operating systems have you tested?

- [ ] macOS
- [X] Windows
- [ ] Linux

**Comment by Tony133** [comment]
Hi @siamahnaf198,  for this type of question try to use the support [discord](https://discord.com/invite/G7Qnnhy) channel

The NestJS Core team uses GitHub to track bug reports, feature requests, and regressions.

**Comment by siamahnaf198** [comment]
Yes, I already asked my questions in discord, But no one is answering me?

**Comment by siamahnaf198** [comment]
That's why I asked my questions here.

Can you help me please

**Comment by siamahnaf198** [comment]
Even I asked questions in Stack Overflow but getting no answer- https://stackoverflow.com/questions/72585276/how-to-create-two-or-more-resolve-field-in-resolve-decorator-in-nestjs

**Comment by NeutronScott12** [comment]
Why not make them separate, whats the reason for trying to mix them.

**Comment by kamilmysliwiec** [comment]
Please, use our [Discord](https://discord.gg/G7Qnnhy) channel (support) for such questions. We are using GitHub to track bugs, feature requests, and potential improvements.
